app/api/route.js

```diff
@@ -0,0 +1,298 @@
+    return mapping;
+  }, {});
+  return author_ids.map((author_id) =>
+    author_id in authorIdToBook ? authorIdToBook[author_id] : []
+  );
+});
+
+const resolvers = {
+  Query: {
+    books: async (_, { page = 0, size = 5, searchTitle = null }) => {
+      const offset = page * size;
+      const where = searchTitle
+        ? {
+            title: {
+              [Op.iLike]: `%${searchTitle}%`,
+            },
+          }
+        : {};
+
+      const { rows, count } = await Book.findAndCountAll({
+        where,
+        limit: size,
+        offset: offset,
+      });
+
+      const validatedBooks = await Promise.all(
+        rows.map(async (book) => ({
+          ...book.toJSON(),
+          profilePhotoUrl: await validateImageUrl(book.profilePhotoUrl, "book"),
+        }))
+      );
+
+      return {
+        books: validatedBooks,
+        totalCount: count,
+        hasMore: offset + rows.length < count,
+      };
+    },
+    authors: async (_, { page = 0, size = 5, searchName = null }) => {
+      const offset = page * size;
+      const where = searchName
+        ? {
+            name: {
+              [Op.iLike]: `%${searchName}%`,
+            },
+          }
+        : {};
+
+      const { rows, count } = await Author.findAndCountAll({
+        where,
+        limit: size,
+        offset: offset,
+      });
+
+      const validatedAuthors = await Promise.all(
+        rows.map(async (author) => ({
+          ...author.toJSON(),
+          profilePhotoUrl: await validateImageUrl(
+            author.profilePhotoUrl,
+            "author"
+          ),
+        }))
+      );
+
+      return {
+        authors: validatedAuthors,
+        totalCount: count,
+        hasMore: offset + rows.length < count,
+      };
+    },
+    book: async (parent, { id }, context, info) => {
+      const book = await Book.findByPk(id);
+      if (!book) {
+        throw new Error("Book not found");
+      }
+      return book;
+    },
+    author: async (parent, args, context, info) => {
+      return Author.findByPk(args.id);
+    },
+    allAuthors: async () => {
+      return await Author.findAll();
+    },
+  },
+  Book: {
+    author: async (parent, args, context, info) => {
+      return batchAuthors.load(parent.author_id);
+    },
+  },
+  Author: {
+    books: async (parent, args, context, info) => {
+      return batchBooks.load(parent.id);
+    },
+  },
+  Mutation: {
+    addBook: async (parent, args, context, info) => {
+      return await Book.create(args, { returning: true });
+    },
+    addAuthor: async (parent, args, context, info) => {
+      const result = await Author.create(args, { returning: true });
+      return result;
+    },
+    updateBook: async (parent, args, context, info) => {
+      if (args.publishedAt) {
+        args.publishedAt = moment(args.publishedAt);
+      }
+      return (
+        await Book.update(args, {
+          where: { id: args.id },
+          returning: true,
+        })
+      )[1][0];
```

Comment:
[1][0] seems very odd and not readable, what is this? Possible to make it more readable?